packages/react-storage/src/components/StorageManager/hooks/useUploadFiles/__tests__/useUploadFiles.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { renderHook } from '@testing-library/react-hooks';
+
+import { Storage, UploadTask } from '@aws-amplify/storage';
+
+import { FileStatus, StorageFile } from '../../../types';
+import { useUploadFiles, UseUploadFilesProps } from '../useUploadFiles';
+import { waitFor } from '@testing-library/react';
+
+jest.mock('@aws-amplify/storage');
+
+const mockUploadingFile: StorageFile = {
+  id: 'uploading',
+  status: FileStatus.UPLOADING,
+  progress: 0,
+  error: '',
+  isImage: false,
+  key: '',
+};
+
+const imageFile = new File(['hello'], 'hello.png', { type: 'image/png' });
+
+const mockQueuedFile: StorageFile = {
+  id: 'queued',
+  status: FileStatus.QUEUED,
+  progress: 0,
+  error: '',
+  isImage: false,
+  key: 'key',
+  file: imageFile,
+};
+
+const mockSetUploadingFile = jest.fn();
+const mockSetUploadProgress = jest.fn();
+const mockSetUploadSuccess = jest.fn();
+const mockOnUploadError = jest.fn();
+const props: Omit<UseUploadFilesProps, 'files'> = {
+  accessLevel: 'public',
+  maxFileCount: 2,
+  setUploadingFile: mockSetUploadingFile,
+  setUploadProgress: mockSetUploadProgress,
+  setUploadSuccess: mockSetUploadSuccess,
+  onUploadError: mockOnUploadError,
+};
+
+const storageOutput: UploadTask = {
+  resume: jest.fn(),
+  pause: jest.fn(),
+  percent: 100,
+  isInProgress: false,
+};
+
+describe('useUploadFiles', () => {
+  afterEach(() => jest.clearAllMocks());
+
+  it('should upload all queued files', async () => {
+    (Storage.put as jest.Mock).mockResolvedValue(storageOutput);
+    renderHook(() =>
+      useUploadFiles({ ...props, files: [mockUploadingFile, mockQueuedFile] })
+    );
+
+    expect(mockSetUploadingFile).toHaveBeenCalledTimes(1);
+    expect(mockSetUploadingFile).toHaveBeenCalledWith({
+      id: mockQueuedFile.id,
+    });
+    expect(mockSetUploadingFile).not.toHaveBeenCalledWith({
+      id: mockUploadingFile.id,
+    });
+
+    await waitFor(() => {
```

Comment:
Believe you can use `waitForNextUpdate` returned from the results of `renderHook` to remove the need for `waitFor`

Reply:
I tried that and didn't work. could try again.

Reply:
Meh, not super important IMO. Probably should have prefixed the comment with `Nit:`